sycl/include/CL/sycl/detail/context_impl.hpp

```diff
@@ -28,6 +28,24 @@ class device;
 namespace detail {
 class context_impl {
 public:
+  template <typename T>
```

Comment:
This class looks like very generic helper which can be used for guarding other entities. Can't we move it to some utils?

Reply:
Moved to distinct hpp file in sycl/inlcude/CL/sycl/detail/